test/unit/server/soroban/assembled_transaction_test.js

```diff
@@ -0,0 +1,123 @@
+      resultXdr: "AAAAAAAAiRkAAAAAAAAAAQAAAAAAAAAaAAAAAAAAAAA=",
+      resultMetaXdr: "AAAAAwAAAAAAAAACAAAAAwAAQowAAAAAAAAAABHCklg6riUqP9F21Lt2zdyIZIx9lSn7t3jGCD2+w3upAAAAF0h1Pp0AAAvWAAAAAgAAAAAAAAAAAAAAAAAAAAABAAAAAAAAAAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAMAAAAAAAAMMQAAAABmTz9yAAAAAAAAAAEAAEKMAAAAAAAAAAARwpJYOq4lKj/RdtS7ds3ciGSMfZUp+7d4xgg9vsN7qQAAABdIdT6dAAAL1gAAAAMAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAADAAAAAAAAQowAAAAAZk919wAAAAAAAAABAAAACAAAAAMAAAwrAAAACc4pIDe7y0sRFHAghrdpB7ypfj4BVuZStvX4u0BC1S/YAAANVgAAAAAAAAABAABCjAAAAAnOKSA3u8tLERRwIIa3aQe8qX4+AVbmUrb1+LtAQtUv2AAAQ7cAAAAAAAAAAwAADCsAAAAJikpmJa7Pr3lTb+dhRP2N4TOYCqK4tL4tQhDYnNEijtgAAA1WAAAAAAAAAAEAAEKMAAAACYpKZiWuz695U2/nYUT9jeEzmAqiuLS+LUIQ2JzRIo7YAABDtwAAAAAAAAADAAAMMQAAAAlT7LdEin/CaQA3iscHqkwnEFlSh8jfTPTIhSQ5J8Ao0wAADVwAAAAAAAAAAQAAQowAAAAJU+y3RIp/wmkAN4rHB6pMJxBZUofI30z0yIUkOSfAKNMAAEO3AAAAAAAAAAMAAAwxAAAACQycyCYjh7j9CHnTm9OKCYXhgmXw6jdtoMsGHyPk8Aa+AAANXAAAAAAAAAABAABCjAAAAAkMnMgmI4e4/Qh505vTigmF4YJl8Oo3baDLBh8j5PAGvgAAQ7cAAAAAAAAAAgAAAAMAAEKMAAAAAAAAAAARwpJYOq4lKj/RdtS7ds3ciGSMfZUp+7d4xgg9vsN7qQAAABdIdT6dAAAL1gAAAAMAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAIAAAAAAAAAAAAAAAAAAAADAAAAAAAAQowAAAAAZk919wAAAAAAAAABAABCjAAAAAAAAAAAEcKSWDquJSo/0XbUu3bN3IhkjH2VKfu3eMYIPb7De6kAAAAXSHWDiQAAC9YAAAADAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAAAAAAAAAwAAAAAAAEKMAAAAAGZPdfcAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAA",
+      ledger: 17036,
+      createdAt: "1716483575",
+    };
+    this.axiosMock
+      .expects("post")
+      .withArgs(
+        serverUrl,
+        sinon.match({
+          jsonrpc: "2.0",
+          id: 1,
+          method: "simulateTransaction",
+        })
+      )
+      .returns(Promise.resolve({ data: { result: simulateTransactionResponse } }));
+    this.axiosMock
+      .expects("post")
+      .withArgs(serverUrl, 
+        sinon.match({
+          jsonrpc: "2.0",
+          id: 1,
+          method: "getTransaction",
+        })
+      )
+      .returns(Promise.resolve({ data: { result: getTransactionResponse } }));
+
+    this.axiosMock
+      .expects("post")
+      .withArgs(serverUrl, 
+        sinon.match({
+          jsonrpc: "2.0",
+          id: 1,
+          method: "sendTransaction",
+        })
+      )
+      .returns(Promise.resolve({ data: { result: sendTransactionResponse } }));
+
+  contract.AssembledTransaction.buildFootprintRestoreTransaction(
+    options,
+    restoreTxnData,
+    new StellarSdk.Account(
+      "GBZXN7PIRZGNMHGA7MUUUF4GWPY5AYPV6LY4UV2GL6VJGIQRXFDNMADI",
+      "1",
+    ),
+    52641,
+  )
+    .then((txn) => txn.signAndSend({force: true, ...wallet,
+      updateTimeout: false
+    }))
```

Comment:
Yes, it turns out that if force is false, the `get simulationData()` will be called which will always fail with       `Expected an invocation simulation, but got no 'result' field. Simulation: ${JSON.stringify(
          simulation,`

Reply:
Oh, what an unexpected workaround! Is this something that's worth fixing? If `buildRestoreFootprintWhatever` is going to stop being public, then we probably don't need to bother fixing it, since no one will use it the same way as the test. You don't need the `force: true` in any other situation, do you?